### DISCUSSION TITLE
Dudas conceptuales de algunos Patrones de Diseño

### DISCUSSION THREAD

**Discussion by neverkas** [dds-jv/foro#189] [Consultas de objetos] [answered]
En una parte menciona  lo siguiente **"Es común que un factory method sea un hook-method de algún template-method"**

1. El factory-method (Ej. crearPanificacion) puede ser un hook-method porque la instancia a la clase que devuelve tiene varios pasos (mezclar,amasar,..) ? y cuando dice de algún template-method es porque este otro tiende a reutilizar los pasos?

2. La diferencia entre los patrones *Strategy* y *Method Template*, el primero tendería a ser un solo problema, con distintas soluciones (algoritmos) y el segundo tiene varios problemas (pasos) que son redefinidos por subclases (distintas implementaciones de cada paso) ?

**Comment by neverkas** [comment, marked as answer]
Buenas, creo que resolví mis dudas con el libro, pero igual dejo las preguntas recortadas
Gracias,

**Reply by gastonprieto** [reply]
> El factory-method (Ej. crearPanificacion) puede ser un hook-method porque la instancia a la clase que devuelve tiene varios pasos (mezclar,amasar,..) ? y cuando dice de algún template-method es porque este otro tiende a reutilizar los pasos?

El hook method sería el paso a definir en cada subclase, en español sería algo como método donde enganchas (hook) el comportamiento de la super clase con la subclase 

>La diferencia entre los patrones Strategy y Method Template, el primero tendería a ser un solo problema, con distintas soluciones (algoritmos) y el segundo tiene varios problemas (pasos) que son redefinidos por subclases (distintas implementaciones de cada paso) ?

claro, en el 2do el algoritmo es como el mismo, pero hay una parte que se completa en las subclases. 

Y una solución podría tener varios patrones (**o ninguno**), por ejemplo podrías usar en Prenda un Strategy y que la jerarquía de la estrategia sea una herencia donde haya un template.

**Reply by gastonprieto** [reply]
te sirvio @neverkas ?
